### PR TITLE
fix smarty deprecation

### LINKED ISSUE #32966: smarty user deprecation notices about php function 'implode'
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

I'm on a BO with debug enabled, I then edit a faceted search category. 
This is the error message I get.

I get it only when the smarty cache is empty. It does not reproduce on reload of the page.

Related to https://github.com/PrestaShop/PrestaShop/issues/32703

### Expected behavior

There should no usage of deprecated smarty features.

### Steps to reproduce

On an installation with debug enabled (`define('_PS_MODE_DEV_', true);`) and ps_facetedsearch installed

- log into BO
- go to ps_facetedsearch configuration
- in "Filter models" click on "Create model" or edit an existing model
- Notice the error in a modal.
- If needed, empty the folder `var/cache`

### PrestaShop version(s) where the bug happened

8.0.4

### PHP version(s) where the bug happened

8.0

### If your bug is related to a module, specify its name and its version

ps_facetedsearch (version not important, bug is in core)

### Your company or customer's name goes here (if applicable).

Société Biblique de Genève

### PR DESCRIPTION
**Pull request by lmeyer1**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | fixing smarty deprecation for `\|implode` as modifier
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Before we get a smarty deprecation error. After this is fixed 
| Fixed ticket?     | Fixes #32966
| Related PRs       | none
| Sponsor company   | Société Bilique de Genève

### CODE REVIEW

**Review by mflasquin — DISMISSED** [review]

**Review by PululuK — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Review by FabienPapet — DISMISSED** [review]

**Review by MhiriFaten — COMMENTED** [review]
Hello @lmeyer1 ,
This PR is on the develop branch, while from the beginning the issue related to this PR exists only on 8.0.x
So could you change the PR branch !

**Comment by jolelievre** [comment]
@lmeyer1 it's weird we already handled this modifier here https://github.com/PrestaShop/PrestaShop/blob/5339457224951b8b1ebef2c40c308dd84095d179/config/smarty.config.inc.php#L114 in theory? Are you sure you tested your issue with the latest develop version?

And it was also fixed on 8.0.x theoretically, but not released yet as a patch I believe https://github.com/PrestaShop/PrestaShop/blob/8.0.x/config/smarty.config.inc.php#L116 So same question did you test with the latest 8.0.x version from the branch?

**Comment by kpodemski** [comment]
@jolelievre well, this changes from using a modifier to using a function directly, not sure if it will be allowed in the future tho 🤔

**Comment by lmeyer1** [comment]
@jolelievre 
Hello, I tested with the latest released version 8.0.4.
Sorry, I don't have time to spend to test develop versions. I'm dedicated to get our production version running smoothly.

File: admin-dev/themes/default/template/controllers/groups/helpers/tree/tree_categories.tpl [hunk 1]
```diff
@@ -82,7 +82,7 @@
 		$("#{$id|escape:'html':'UTF-8'}").tree("collapseAll");
 
 		{if isset($selected_categories)}
-			{assign var=imploded_selected_categories value='","'|implode:$selected_categories}
```
Comment (lmeyer1) [inline admin-dev/themes/default/template/controllers/groups/helpers/tree/tree_categories.tpl:85]: The code is not equivalent before and after. We must use `implode('","', $selected_categories);`
Nobody noticed it 😄

**Comment by SharakPL** [comment]
Hasn't this been fixed already in https://github.com/PrestaShop/PrestaShop/pull/33136 ?

**Comment by lmeyer1** [comment]
@SharakPL  Yes, it is.

**Pull request closed without merging**